src/main/java/hybrid/it/internship/library/service/impl/RentServiceImpl.java

```diff
@@ -21,4 +42,89 @@ public boolean existsByBookId(Long bookId) {
     public boolean existsByUserId(Long userId) {
         return rentRepository.existsByUserId(userId);
     }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<RentDTO> getOverdueRents() {
+
+        List<RentDTO> rentsDTOs = rentMapper.toDTO(rentRepository.findAll());
```

Comment:
potentially low performant operation, If the number of rents is a high number,
Consider adding an addtional field in rent table (returnByDate for example) which could be calculated in moment of renting a book. Then finding overdue books could be done by JPA query where current_date is after returnByDate